sync/sync_head.go

```diff
@@ -22,6 +22,7 @@ func (s *Syncer[H]) Head(ctx context.Context, _ ...header.HeadOption[H]) (H, err
 	if isRecent(sbjHead, s.Params.blockTime, s.Params.recencyThreshold) {
 		return sbjHead, nil
 	}
+	s.metrics.laggingNetworkHead(s.ctx)
```

Comment:
We should move under the lock so we track this only once a time. There is not much we get from tracking this for every routine